contracts/src/token/erc1155/extensions/supply.rs

```diff
@@ -0,0 +1,824 @@
+
+    /// Override of [`Erc1155::_update`] that updates the supply of tokens.
+    ///
+    /// # Arguments
+    ///
+    /// * `&mut self` - Write access to the contract's state.
+    /// * `from` - Account of the sender.
+    /// * `to` - Account of the recipient.
+    /// * `token_ids` - Array of all token id.
+    /// * `values` - Array of all amount of tokens to be supplied.
+    ///
+    /// # Errors
+    ///
+    /// If length of `ids` is not equal to length of `values`, then the
+    /// error [`erc1155::Error::InvalidArrayLength`] is returned.
+    /// If `value` is greater than the balance of the `from` account,
+    /// then the error [`erc1155::Error::InsufficientBalance`] is returned.
+    ///
+    /// NOTE: The ERC-1155 acceptance check is not performed in this function.
+    /// See [`Self::_update_with_acceptance_check`] instead.
+    ///
+    /// # Events
+    ///
+    /// Emits a [`erc1155::TransferSingle`] event if the arrays contain one
+    /// element, and [`erc1155::TransferBatch`] otherwise.
+    ///
+    /// # Panics
+    ///
+    /// If updated balance and/or supply exceeds `U256::MAX`, may happen during
+    /// the `mint` operation.
+    fn _update(
+        &mut self,
+        from: Address,
+        to: Address,
+        token_ids: Vec<U256>,
+        values: Vec<U256>,
+    ) -> Result<(), erc1155::Error> {
+        self.erc1155._update(from, to, token_ids.clone(), values.clone())?;
+
+        if from.is_zero() {
+            for (&token_id, &value) in token_ids.iter().zip(values.iter()) {
+                let total_supply =
+                    self.total_supply(token_id).checked_add(value).expect(
+                        "should not exceed `U256::MAX` for `_total_supply`",
+                    );
+                self._total_supply.setter(token_id).set(total_supply);
+            }
+
+            let total_mint_value = values.iter().sum();
+            let total_supply_all =
+                self.total_supply_all().checked_add(total_mint_value).expect(
+                    "should not exceed `U256::MAX` for `_total_supply_all`",
+                );
+            self._total_supply_all.set(total_supply_all);
+        }
+
+        if to.is_zero() {
+            for (token_id, &value) in token_ids.into_iter().zip(values.iter()) {
+                /*
+                SAFETY: Overflow not possible:
+                values[i] <= balance_of(from, token_ids[i]) <= total_supply(token_ids[i])
+                 */
+                self._total_supply.setter(token_id).sub_assign_unchecked(value);
+            }
+
+            let total_burn_value: U256 = values.into_iter().sum();
+            /*
+            SAFETY: Overflow not possible:
+            total_burn_value = sum_i(values[i]) <= sum_i(total_supply(ids[i])) <= total_supply_all
+             */
+            let total_supply_all =
+                self._total_supply_all.get() - total_burn_value;
+            self._total_supply_all.set(total_supply_all);
```

Comment:
I believe you can use here also `sub_assign_unchecked`

Reply:
Implemented the relevant trait for `StorageUint` (https://github.com/OpenZeppelin/rust-contracts-stylus/pull/418/commits/2436791b11b1498e1f3c344b0b39a99d14a1777f) and was then able to do as suggested